core/src/main/java/org/apache/calcite/plan/volcano/VolcanoPlanner.java

```diff
@@ -1081,60 +1072,6 @@ private RelNode changeTraitsUsingConverters(
     return converted;
   }
 
-  /**
-   * Converts traits using well-founded induction. We don't require that
-   * each conversion preserves all traits that have previously been converted,
-   * but if it changes "locked in" traits we'll try some other conversion.
-   *
-   * @param rel                         Relational expression
-   * @param allowInfiniteCostConverters Whether to allow infinite converters
-   * @param toTraits                    Target trait set
-   * @param usedTraits                  Traits that have been locked in
-   * @return Converted relational expression
-   */
-  private RelNode completeConversion(
-      RelNode rel,
-      boolean allowInfiniteCostConverters,
-      RelTraitSet toTraits,
-      Expressions.FluentList<RelTraitDef> usedTraits) {
-    if (true) {
-      return rel;
```

Comment:
Please hold on for the fix, i'm curious if this logic is left there intentionally. @julianhyde Can you please take a look at this code ? There are many places in Calcite with the special "useless" `if(true)` logic, for me, it makes the logic more clear because it illustrates how the logic evolves.

Reply:
I already replied in the jira case